simpleui-server/src/SuiZMQ.ts

```diff
@@ -0,0 +1,150 @@
+import { SuiData } from './sui_data';
+import { Queue } from './queue';
+
+
+export class ZMQ_Socker_Wrapper {
+    hostname: string;
+    timeout: number;
+    port: number;
+    socket: any;
+    http_queue: Queue;
+    
+
+    constructor(hostname: string='svcmachineapps', port: number, timeout: number=500) {
+        this.hostname = hostname;
+        this.timeout = timeout;
+        this.port = port;
+        this.http_queue = new Queue();
+
+        try {
+            this.socket = _zmq.socket('req');
+            this.socket.connect_timeout = timeout;
+            this.connect(this.port);
+
+            this.socket.on('message', (msg) => {
+                const raw_zmq_data = msg.toString();
+                const zmq_data = SuiData.addXmlStatus(raw_zmq_data);
+                Logger.log(LogLevel.DEBUG, `Recieved ZMQ message at port ${this.port}: ${zmq_data.substring(0, 105)}`);
+                // get response + request from queue
+                let [res, req] = this.http_queue.dequeue();
+                // send response
+                SuiData.sendResponse(req, res, zmq_data);
+            });
+
+            this.socket.on('error', (zmqErr) => {
+                const zmq_data = ServerUtil.getServerError('ZMQ_ERROR', '{{ERROR}}', zmqErr);
+                Logger.log(LogLevel.ERROR, `ZMQ socket at port ${this.port} got error: ${zmqErr}`);
+                // get response + request from queue
+                let [res, req] = this.http_queue.dequeue();
+                // send response
+                SuiData.sendResponse(req, res, zmq_data);
+            });
+
+        } catch (err) {
+            Logger.log(LogLevel.ERROR, `Could not create ZMQ socket at port ${this.port} got error: ${err}`)
+        }
+    }
+
+    connect(port: number) {
+        try {
+            Logger.log(LogLevel.VERBOSE, `ZMQ connecting to --->  tcp://${this.hostname}:${port}`)
+            this.socket.connect(`tcp://${this.hostname}:${port}`)
+
+        } catch (e) {
+            Logger.log(LogLevel.ERROR, `Could not connect to tcp://${this.hostname}:${port}`);
+        }
+    }
+
+    send(msg: string) {
+        try{
+            this.socket.send(msg);
+        } catch (err) {
+            Logger.log(LogLevel.ERROR, `could not send zmq message:\n${msg}`)
+        }
+    }
+
+    close() {
+        if (this.socket.closed === false) {
+            this.socket.close();
+        }
+    }
+
+    set_timeout(ms: number) {
+        try {
+            this.socket.connect_timeout = ms;
+        } catch (err) {
+            Logger.log(LogLevel.ERROR, `COULD NOT SET ZMQ TIMEOUT ${ms}`);
+        }
+    }
+}
+
+
+/**
+ * This class handles the creation of the zmq map 
+ * and the destruction of the sockets
+ */
+export class zmq_wrapper {
+    socket_map: Map<number, ZMQ_Socker_Wrapper>;
+
+    constructor(port_list: Array<number>) {
+        this.socket_map = new Map();
+
+        for (let port_indx = 0; port_indx < port_list.length; port_indx++) {
```

Comment:
A few things here:

1. If you're going to use a raw for loop you're best off just using `i` for the loop variable cause everyone knows it's just an index. Sometimes I use `i_port` or whatever if there are other loop variables or indices involved.
2. You actually don't need a raw for loop at all, you can just call port_list.forEach((portNumber)=>{//stuff})

Reply:
I'll keep that in mind about naming indices . I ended up swapping the raw for loop out for `forEach()`